src/main/java/com/three_tech_solutions/slot_app/data/repositories/SlotRepository.java

```diff
@@ -18,4 +19,12 @@ SELECT COUNT(s) > 0
         AND :time <  s.endTime
     """)
     boolean existsWithinRange(@Param("time") LocalTime time, @Param("dayOfWeek") DayOfWeek dayOfWeek);
+
+    @Query("""
+            SELECT s FROM Slot s
+            WHERE s.user.id = :userId
+            AND s.dayOfWeek = :dayOfWeek
+            ORDER BY s.startTime ASC
+            """)
+    List<Slot> findAllByUserIdAndDayOfWeekOrdered(UUID userId, DayOfWeek dayOfWeek);
```

Comment:
Esto podriamos modificarlo asi si directamente recibimos el user


```suggestion
    @Query("""
            SELECT s FROM Slot s
            WHERE s.user = :user
            AND s.dayOfWeek = :dayOfWeek
            ORDER BY s.startTime ASC
            """)
    List<Slot> findAllByUserIdAndDayOfWeekOrdered(User user, DayOfWeek dayOfWeek);
```

Esto es lo que se suele hacer

Reply:
pero recibimos el id del user. En el primero comentario que me hiciste en el userService, yo tengo esto:
        return slotService.getSlotsByDayOfWeek(getUserByIdOrThrowException(userId).getId(), dayOfWeek);
        y me decías que deberia cambiarlo por esto:
        return slotService.getSlotsByDayOfWeek(userId, dayOfWeek);
        de las dos formas tengo el id, a no ser que haga 
         return slotService.getSlotsByDayOfWeek(getUserByIdOrThrowException(userId), dayOfWeek);

Reply:
Hacé el cambio de recibir el user en el repositorio y dejalo con esto en el service

`return slotService.getSlotsByDayOfWeek(getUserByIdOrThrowException(userId), dayOfWeek);`

De esta manera tambien haces la validacion de que exista el user